File: 프로그래머스/3/42861. 섬 연결하기/섬 연결하기.js
function solution(N, costs) {
    var answer = 0;
    costs.sort((a,b)=>a[2]-b[2])
    const parents = Array.from({length:N}).fill(0).map((_,idx)=>idx)
    function find(x){
        if(parents[x]!=x){
            parents[x]=find(parents[x])
        }
        return parents[x]
    }
    function union(a,b){
        a = find(a)
        b = find(b)
        if(a!=b){
            parents[b] = parents[a]
            return true
        }
        return false
    }
    for (let cost of costs){
        const [a,b,c] = cost
        // console.log(a,b, parents[a],parents[b])
        if(union(a,b)){
            answer+=c
        }
    }
    return answer;
}

File: 프로그래머스/3/42861. 섬 연결하기/섬 연결하기.py
### 유니온 파인드로 풀어보자
def find_parent(x, parent):
    if x!=parent[x]:
        x = find_parent(parent[x],parent)
    return parent[x]

def union(a,b,parent):
    a=find_parent(a,parent)
    b=find_parent(b,parent)
    if a<b:
        parent[b]=a
    else:
        parent[a]=b

def solution(n, costs):
    answer = 0
    costs.sort(key=lambda x:(x[2]))
    parent=[i for i in range(n)]
    for a,b,c in costs:
        if find_parent(a, parent)!=find_parent(b,parent):
            union(a,b,parent)
            answer+=c
            
    return answer